lib/src/components/AbstractDialog/__tests__/AbstractDialog.test.tsx

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React from 'react';
+
+import { render, fireEvent, waitFor, createEvent } from '@testing-library/react';
+import { act } from 'react-dom/test-utils';
+import { screen } from '@testing-library/dom';
+import userEvent from '@testing-library/user-event';
+
+import { rtlRender, rtlRenderElement } from '../../../__testutils__/rtlUtils';
+
+import AbstractDialog, { Props } from '../AbstractDialog';
+import NxButton from '../../NxButton/NxButton';
+import NxTooltip from '../../NxTooltip/NxTooltip';
+import useToggle from '../../../util/useToggle';
+
+describe('AbstractDialog', function() {
```

Comment:
I tend to think we don't want unit tests for `AbstractDialog` itself.  It isn't exported outside of the library; its entire existence is an implementation detail of `NxModal` and `NxDrawer`, and some of the things being tested here around event handling seem too much like implementation details.  I'd rather just have thorough tests of the two external-facing components.

Reply:
> I tend to think we don't want unit tests for `AbstractDialog` itself. It isn't exported outside of the library; its entire existence is an implementation detail of `NxModal` and `NxDrawer`, and some of the things being tested here around event handling seem too much like implementation details. I'd rather just have thorough tests of the two external-facing components.

That makes sense, we removed it here: d89e53eb5c06d82ee732d20acc0f0f7eb7f0cb83

Reply:
> I tend to think we don't want unit tests for `AbstractDialog` itself. It isn't exported outside of the library; its entire existence is an implementation detail of `NxModal` and `NxDrawer`, and some of the things being tested here around event handling seem too much like implementation details. I'd rather just have thorough tests of the two external-facing components.

Fixed here: d89e53eb5c06d82ee732d20acc0f0f7eb7f0cb83